### DISCUSSION TITLE
Is there any a way to make `./app --help` the default behavior when there's no argument passed?

### DISCUSSION THREAD

**Discussion by ghost** [pacak/bpaf#219] [Q&A] [answered]
Hi :smiley: ! Thanks for making/maintaining this library.

I've been looking for this in the doc, but sadly I couldn't find it or missed it somewhere (most likely).

I wonder if we could invoke the program without arguments and let it show us the help message by default, the same if you pass `--help` to the program. For reference, please see Haskell's [optparse-applicative](https://hackage.haskell.org/package/optparse-applicative-0.18.1.0/docs/Options-Applicative.html#v:showHelpOnEmpty) package with `showHelpOnEmpty` preference enabled.

This is what I have in mind:
```rust
    // top-level options parser
    construct!([tui_cmd, install_cmd])
        .to_options()
        .show_help_on_empty()
        .version("v0.1.0")
        .run()
```

Output example:

```
$ ./app
Usage: COMMAND ...

Available options:
    -h, --help     Prints help information
    -V, --version  Prints version information

Available commands:
    tui      Show TUI interface
    install  Install
```

Thanks!

**Comment by pacak** [comment, marked as answer]
> Thanks for making/maintaining this library.

Thank you for the kind words :)

> I wonder if we could invoke the program without arguments and let it show us the help message by default, the same if you pass `--help` to the program.

Not in the current version, but it might change in the future given enough demand. For now you can do something like this to achieve the same results:

https://github.com/pacak/bpaf/discussions/186

**Reply by ghost** [reply]
Thank you!

**Comment by pacak** [comment]
Since 0.9.1 you can use `OptionParser::fallback_to_usage`, see https://github.com/pacak/bpaf/blob/bc42de419b15aba586bd4cd2f0b7874f3f915d38/examples/basic.rs
